report-microservice/src/main/java/com/epamlearning/reportmicroservice/services/impl/TrainerSummaryServiceImpl.java

```diff
@@ -20,42 +27,62 @@
+            years.add(yearMonth.getYear());
 
-  @Override
-  public TrainerSummary calculate(String username) {
-    List<TrainerWorkload> trainerWorkloads = trainerWorkloadService.getAllByUsername(username);
-    if(trainerWorkloads.isEmpty()) {
-      throw new NotFoundException("No trainer workload found for username to calculate summary: " + username);
+            BigDecimal duration = trainerWorkload.getTrainingDuration();
+
+            monthlySummaryMap.computeIfAbsent(year, k -> new HashMap<>())
+                    .merge(month, duration, BigDecimal::add);
+        }
+
+        return createTrainerSummary(trainerWorkloads.getFirst(), years, monthlySummaryMap);
     }
-    Map<String, Map<String, BigDecimal>> monthlySummaryMap = new HashMap<>();
-    HashSet<Integer> years = new HashSet<>();
-    for(TrainerWorkload trainerWorkload : trainerWorkloads) {
-      LocalDate trainingDate = trainerWorkload.getTrainingDate();
-      YearMonth yearMonth = YearMonth.from(trainingDate);
-      String month = yearMonth.getMonth().toString();
-      String year = String.valueOf(yearMonth.getYear());
-      years.add(yearMonth.getYear());
-
-      BigDecimal duration = trainerWorkload.getTrainingDuration();
-
-      monthlySummaryMap.computeIfAbsent(year, k -> new HashMap<>())
-          .merge(month, duration, BigDecimal::add);
+
+    public TrainerSummary createTrainerSummary(TrainerWorkload trainerWorkload,
+                                               HashSet<Integer> years, Map<String,
+            Map<String, BigDecimal>> monthlySummaryMap) {
+        TrainerSummary trainerSummary = new TrainerSummary();
+        trainerSummary.setFirstName(trainerWorkload.getFirstName());
+        trainerSummary.setLastName(trainerWorkload.getLastName());
+        trainerSummary.setStatus(trainerWorkload.isActive());
+        trainerSummary.setUsername(trainerWorkload.getUsername());
+        trainerSummary.setYears(years.stream().sorted().toList());
+        trainerSummary.setMonthlySummary(monthlySummaryMap);
+        return trainerSummary;
     }
 
-    TrainerSummary trainerSummary = new TrainerSummary();
-    trainerSummary.setFirstName(trainerWorkloads.getFirst().getFirstName());
-    trainerSummary.setLastName(trainerWorkloads.getFirst().getLastName());
-    trainerSummary.setStatus(trainerWorkloads.getFirst().isActive());
-    trainerSummary.setUsername(username);
-    trainerSummary.setYears(years.stream().sorted().toList());
-    trainerSummary.setMonthlySummary(monthlySummaryMap);
-    return trainerSummary;
-  }
+    @JmsListener(destination = "trainer-summary-queue")
+    public JmsResponse<Message<TrainerSummary>> getTrainerSummary(@Payload String username, @Header("jms_correlationId") String transactionId) {
+        MDC.put("transactionId", transactionId);
+        Message<TrainerSummary> message = MessageBuilder
+                .withPayload(calculate(username))
+                .setHeader("jms_correlationId", transactionId)
```

Comment:
It's a good practice to put the duplicate value of a string into a constant and use it in the application. it can be as a class variable or in a separate class